editor.planx.uk/src/@planx/components/shared/Schema/model.ts

```diff
@@ -36,70 +36,104 @@ interface ChecklistInput {
 /**
  * As above, we need a simplified validation schema for QuestionsInputs
  */
-const questionInputValidationSchema = (data: QuestionInput) =>
+
+export const questionInputValidationSchema = ({
+  data,
+  required,
+}: FieldValidationSchema<QuestionInput>) =>
   string()
-    .oneOf(data.options.map((option) => option.data.val || option.data.text))
-    .required("Select your answer before continuing");
+    .when([], {
+      is: () => required,
+      then: string().required("Select your answer before continuing"),
+      otherwise: string().notRequired(),
+    })
+    .test("isValidOption", "Invalid selection", (value) => {
+      if (!value) return true;
+
+      // Check validity regardless of "required" status
+      return data.options.some(
+        (option) => value === (option.data.val || option.data.text),
+      );
+    });
+
+/**
+ * Describes the input of function which returns a Yup validation schema for a field
+ */
+export interface FieldValidationSchema<T extends Omit<Field["data"], "fn">> {
+  data: T;
+  required: boolean;
+}
 
-const mapValidationSchema = ({ mapOptions }: MapField["data"]) =>
+export const mapInputValidationSchema = ({
+  data: { mapOptions },
+  required,
+}: FieldValidationSchema<MapInput>) =>
   array()
-    .required()
+    .when([], {
+      is: () => required,
+      then: array().required(),
+      otherwise: array().notRequired(),
+    })
     .test({
       name: "atLeastOneFeature",
       message: `Draw at least one ${
         mapOptions?.drawType?.toLocaleLowerCase() || "feature"
       } on the map`,
       test: (features?: Array<Feature>) => {
-        return Boolean(features && features?.length > 0);
+        if (!required && !features?.length) return true;
+        if (!features || !features.length) return false;
+
+        const isGeoJSON = (input: Feature) => input?.type === "Feature";
+
+        return features.every(isGeoJSON);
       },
     });
 
-export type TextField = {
-  type: "text";
-  data: TextInput & { fn: string };
+type BaseField<T> = {
+  required?: boolean;
+  data: T & { fn: string };
```

Comment:
This is the fundamental change in this PR - 

 - A new `BaseField` interface with all other fields extends
 - `required` is set her, not at a lower level in `data` (more below)
 - `fn` is being made required at this level (no functional change, moving it here simply saves repetition)

**Why are we setting required at the highest level, and not as part of `data`?**
It's set here as we're making the entire "field" (which can consist of multiple inputs) optional. All PlanX components are currently required, and we're very much restricting this change to schema fields only - not their related components. Setting this here hopefully makes this distinction more explicit.